### ISSUE TITLE
Do not attach subscriptions during execution of `createElement`, only during mounting

### ISSUE THREAD

**Issue by Bloomca** [Bloomca/veles#45]
## Description

Right now if you create a Veles node and do nothing with it, the subscription will still exist:

```jsx
const a = (
  <div>
    {someState.useValue()}
  </div>
);
```

And it basically will be never cleaned up, because there is no "unmount" event (since it was never mounted).

## Approach

I think the best approach would be to add `onMount` on every type of Node. We already have `onUnmount`, so expanding it to the mount callbacks should not be hard.

**Comment by Bloomca** [comment]
I discovered that basically every type is affected by this problem. I think I fixed it in `useValue`/`useValueSelector`, and `useValueIterator` should be a pretty similar solution.

That being said, `trackValue` and `trackValueSelector` have an issue with the first call. Currently the first call happens immediately, and there is no way to know at the time whether it is going to be mounted or not.

There are 2 approaches:

- keep it as is clearly marking that behaviour (there are options `skipFirstCall` and `callOnMount` which both solve this issue)
- make `callOnMount` behaviour the default one

I lean to the second one, I don't see much benefit in keeping `onInit` one. It is a big caveat though, to understand that the component's body will be executed even if you don't end up mounting it at all.

Maybe in the future I can refactor so that components are not executed at all until they are ready to be mounted. I think as long as you are aware it should be an okay compromise.
